### PR TITLE
Migrate legacy testsuite to RSpec

### PR DESCRIPTION
**Pull request by ancorgs**
The old legacy testsuite is a blocker for the evolution of YaST and is causing breakage in Travis.

So this pull request migrates those tests to RSpec.

It also fixes some problems in the preexisting RSpec tests, that were wrongly placed and were not loading the source code properly (they needed some extra `require` as a consequence that are now removed).

### CODE REVIEW

File: package/yast2-squid.spec [hunk 1]
```diff
@@ -39,7 +39,6 @@ BuildRequires:  update-desktop-files
 BuildRequires:  yast2 >= 4.1.0
 BuildRequires:  yast2-core-devel
 BuildRequires:  yast2-devtools >= 4.2.2
-BuildRequires:  yast2-testsuite
```
Comment (jreidinger) [inline package/yast2-squid.spec:42]: I think also perl-XML-Writer can be removed.

File: test/complex_include_test.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #! /usr/bin/env rspec
 
 require_relative "./test_helper"
-require_relative "../../src/include/squid/complex"
+require_relative "../src/include/squid/complex"
```
Comment (jreidinger) [inline test/complex_include_test.rb:4]: this really should not be needed. Proper Y2DIR env and Yast.include should just works.

File: test/squid_test.rb [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+      allow(Yast2::SystemService).to receive(:find).with("squid").and_return(service)
+
+      allow(Yast::Mode).to receive(:auto) { auto }
+      allow(Yast::Mode).to receive(:commandline) { commandline }
+
+      allow(squid).to receive(:Abort).and_return(false)
+      allow(squid).to receive(:writeAllSettings).and_return(settings_written)
+      allow(squid).to receive(:writeFirewallSettings).and_return(firewall_settings_written)
+
+      squid.main
```
Comment (jreidinger) [inline test/squid_test.rb:27]: why it is needed only for write and not for all tests?

File: test/squid_test.rb [hunk 4]
```diff
@@ -0,0 +1,785 @@
...
+        let(:service_enabled_on_startup) { false }
+
+        it "calls to #EnableService" do
+          expect(squid).to receive(:DisableService)
+
+          squid.Write
+        end
+      end
+
+      context "and only is writing settings" do
```
Comment (jreidinger) [inline test/squid_test.rb:70]: I expect here to have let with write_only set to true, even if global one is above. It is just more explicit what is happening in tests and allows to change global mock value without breaking tests.

File: test/squid_test.rb [hunk 5]
```diff
@@ -0,0 +1,785 @@
...
+          "case_sensitive" => true, "max" => "10080", "min" => "1440",
+          "percent" => "20", "regexp"=>"^ftp:"
+        },
+        {
+          "case_sensitive" => false, "max" => "1440", "min" => "1440",
+          "percent" => "0", "regexp"=>"^gopher:"
+        },
+        {
+          "case_sensitive" => true, "max" => "4320", "min" => "0",
+          "percent" => "20", "regexp"=>"."
```
Comment (jreidinger) [inline test/squid_test.rb:205]: For me both of the test cases ( this and previous ) will be more readable if pairs of data value and result value was close together, so I can see what changes in what.

So something like:
```
PATTERNS = {
  ["^ftp:", "1440", "20%", "10080"] => {
          "case_sensitive" => true, "max" => "10080", "min" => "1440",
          "percent" => "20", "regexp"=>"^ftp:"
        },
["-i", "^gopher:", "1440", "0%", "1440"] => {
          "case_sensitive" => false, "max" => "1440", "min" => "1440",
          "percent" => "0", "regexp"=>"^gopher:"
        },
...
}

      allow(Yast::SCR).to receive(:Read).with(path_matching(/\.squid\..*refresh_pattern/)).and_return(PATTERNS.keys)
      squid.readRefreshPatterns
      expect(squid.refresh_patterns).to eq PATTERNS.values
```

File: test/squid_test.rb [hunk 6]
```diff
@@ -0,0 +1,785 @@
...
+        { "name" => "Safe_ports", "options" => ["21"], "type" => "port" },
+        { "name" => "Safe_ports", "options" => ["443"], "type" => "port" },
+        { "name" => "Safe_ports", "options" => ["70"], "type" => "port" },
+        { "name" => "Safe_ports", "options" => ["210"], "type" => "port" },
+        { "name" => "Safe_ports", "options" => ["1025-65535"], "type" => "port" },
+        { "name" => "Safe_ports", "options" => ["280"], "type" => "port" },
+        { "name" => "Safe_ports", "options" => ["488"], "type" => "port" },
+        { "name" => "Safe_ports", "options" => ["591"], "type" => "port" },
+        { "name" => "Safe_ports", "options" => ["777"], "type" => "port" },
+        { "name" => "CONNECT", "options" => ["CONNECT"], "type" => "method" }
```
Comment (jreidinger) [inline test/squid_test.rb:255]: and same note here.

**Comment by jreidinger** [comment]
in general impressive work. Only rubocop disagrees :)

File: test/squid_test.rb (hunk 3 above)
Comment (ancorgs) [reply test/squid_test.rb:27]: I didn't write the tests for `Write`. I just moved them from the old location.

File: test/squid_test.rb (hunk 4 above)
Comment (ancorgs) [reply test/squid_test.rb:70]: Same here, I just moved these tests. I didn't even read them. :-)

File: test/complex_include_test.rb (hunk 2 above)
Comment (ancorgs) [reply test/complex_include_test.rb:4]: That's what I thought, but just removing it and using `Yast.include` was not enough. Due to the the way the test it's written other problems arose. I considered it to be kind of out of scope. Migrating the legacy tests is enough, I don't want to also spend time improving the pre-existing RSpec ones.

File: test/squid_test.rb (hunk 5 above)
Comment (ancorgs) [reply test/squid_test.rb]: Changed

File: test/squid_test.rb (hunk 6 above)
Comment (ancorgs) [reply test/squid_test.rb]: Changed as well

File: package/yast2-squid.spec (hunk 1 above)
Comment (ancorgs) [reply package/yast2-squid.spec:42]: Done.

**Pull request merged**